### PR TITLE
feat(dev/plugin): support secrets

### PR DESCRIPTION
**Pull request by mkilpatrick**
This adds support for truly secret environment variables. Before this change we only supported _public_ environment variables that were prefixed with `YEXT_PUBLIC`.

### CODE REVIEW

File: packages/pages/src/util/processEnvVariables.ts [hunk 1]
```diff
@@ -8,20 +8,21 @@ import { loadEnv } from "vite";
...
-    Object.entries(processEnv)
-      .filter(([env]) => env.startsWith(prefix))
-      .map(([key, value]) => [
-        key,
-        stringifyValues ? JSON.stringify(value) : value,
-      ])
-  );
 
-  return processEnv;
+  // If we're development, return all env var keys, otherwise use Vite's default
```
Comment (asanehisa) [inline packages/pages/src/util/processEnvVariables.ts:15]: nit: if we're in development

**Review by asanehisa — APPROVED** [review]
lgtm

File: packages/pages/src/util/processEnvVariables.test.ts [hunk 2]
```diff
@@ -4,30 +4,34 @@ import { processEnvVariables } from "./processEnvVariables.js";
...
 
     expect(Object.keys(env).length).toEqual(1);
     expect(env.YEXT_PUBLIC_KEY).toEqual(`"pk.0123456789"`);
   });
 
-  it("does not stringify values", () => {
-    const env = processEnvVariables("YEXT_PUBLIC", false);
+  it("returns all env vars for development", () => {
+    global.process.env.NODE_ENV = "development";
+    const env = processEnvVariables("YEXT_PUBLIC");
```
Comment (jwartofsky-yext) [inline packages/pages/src/util/processEnvVariables.test.ts:29]: We could have an example with another key, like VITE

**Review by jwartofsky-yext — APPROVED** [review]
lgtm

File: packages/pages/src/util/processEnvVariables.test.ts (hunk 2 above)
Comment (mkilpatrick) [reply packages/pages/src/util/processEnvVariables.test.ts:29]: The key actually doesn't matter in this case as it returns everything. Ack.

**Review by asanehisa — DISMISSED** [review]

**Comment by mkilpatrick** [comment]
I've opened an [issue](https://github.com/vitejs/vite/issues/14929) with VIte re: the Ubuntu failures. Looks like it has to do with loadEnv and an empty string prefix.

**Review by asanehisa — APPROVED** [review]

**Pull request merged**